src/SIL.Harmony.Tests/ModelSnapshotTests.cs

```diff
@@ -33,6 +33,58 @@ public async Task ModelSnapshotShowsMultipleChanges()
         snapshot.LastChange.Should().Be(secondChange.DateTime);
     }
 
+    [Fact]
+    public async Task CanGetWordForASpecificCommit()
+    {
+        var entityId = Guid.NewGuid();
+        var firstCommit = await WriteNextChange(SetWord(entityId, "first"));
+        var secondCommit = await WriteNextChange(SetWord(entityId, "second"));
+        var thirdCommit = await WriteNextChange(SetWord(entityId, "third"));
+        await ClearNonRootSnapshots();
+        var firstWord = await DataModel.GetAtCommit<Word>(firstCommit.Id, entityId);
+        firstWord.Should().NotBeNull();
+        firstWord.Text.Should().Be("first");
+
+        var secondWord = await DataModel.GetAtCommit<Word>(secondCommit.Id, entityId);
+        secondWord.Should().NotBeNull();
+        secondWord.Text.Should().Be("second");
+
+        var thirdWord = await DataModel.GetAtCommit<Word>(thirdCommit.Id, entityId);
+        thirdWord.Should().NotBeNull();
+        thirdWord.Text.Should().Be("third");
+    }
+
+    [Fact]
+    public async Task CanGetWordForASpecificTime()
+    {
+        var entityId = Guid.NewGuid();
+        var firstCommit = await WriteNextChange(SetWord(entityId, "first"));
+        var secondCommit = await WriteNextChange(SetWord(entityId, "second"));
+        var thirdCommit = await WriteNextChange(SetWord(entityId, "third"));
+        await ClearNonRootSnapshots();
```

Comment:
This is supposed to demonstrate that the test isn't relying on any intermediate snapshots, right?
The fact that it's here makes me wonder if the test needs it. If the test needes it, that would sort of mean that the feature doesn't really work, because I don't think snapshots shouldn't affect the DataModel.

If I'm wrong (or you want to keep this), maybe add a code comment explaining why you're calling it.

```suggestion
```

Reply:
that's right, by clearing the snapshots it means that when you call `GetAtTime` it can't just use a snapshot made for that commit it has to rebuild the state from scratch, which requires a different set of filters.